x/erc20/keeper/ibc_callbacks.go

```diff
@@ -89,6 +89,12 @@ func (k Keeper) OnRecvPacket(
 		data.Denom, data.Amount,
 	)
 
+	// If the coin denom starts with `factory/` then it is a token factory coin, and we should not convert it
+	// NOTE: Check https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/ for more information
+	if strings.HasPrefix(data.Denom, "factory/") {
```

Comment:
If this is arriving from ibc. Shouldnt the prefix be "transfer/channel-x/factory/". And this would be detected later by `ibc.IsNativeFromSourceChain(data.Denom)`

Reply:
The `data.Denom` is the direct, non IBC prefixed denom. The `coin.Denom` later has the full IBC denom trace.